### DISCUSSION TITLE
Adding a module from another module

### DISCUSSION THREAD

**Discussion by Nks** [nuxt/nuxt#21774] [Questions] [answered]
Suppose there is a module that has a dependency on another module. For example, the cms package depends on the `@pinia/nuxt` module. According to the documentation, both of these modules should be included in the main nuxt.config.ts, But in this case you lose the meaning of the main module - especially with a large number of sites that will use the same module - you will need to add these modules to each site individually.

Is there any way to connect one module from another? 

I tried something like this, but with no luck, and believe that this should be done differently (like via hooks):

```   
if (!nuxt.options.modules) {
     nuxt.options.modules = [];
}

nuxt.options.modules.concat([ '@pinia/nuxt']);
```

Thanks!

**Comment by Nks** [comment, marked as answer]
It looks like here is the solution:

1. Add dependency to your module. e.g., `pnpm add @pinia/nuxt`
2. Inside your module call `await installModule('@pinia/nuxt');`

**Reply by pi0** [reply]
(important: never forget about `await` for `installModule` 💯 )
